### PR TITLE
EIP-6110: Supply Validator Deposits on Chain

### PR DESCRIPTION
**Pull request by scorbajio**
This PR implements EIP-6110 and addresses issue #3299.

### CODE REVIEW

**Comment by jochem-brouwer** [comment]
I've pointed https://github.com/ethereumjs/ethereumjs-monorepo/pull/3336 towards this branch :)

Before we merge this one into `master`, we should ensure that we can can still use Verkle, and possibly also use Verkle together with 6110 + 7002.

**Comment by holgerd77** [comment]
Ugh.

Now having read this EIP for the first time I’m realizing what a complex beast this is, always considered this more as some „side EIP“. 🤯

@scorbajio Congrats on that, this looks already pretty great! 🤩 Really blown away by the amount of code already touched here!

File: packages/block/src/types.ts [hunk 1]
```diff
@@ -168,12 +177,26 @@ export type BlockBytes =
       WithdrawalsBytes,
       ExecutionWitnessBytes
     ]
+  | [BlockHeaderBytes, TransactionsBytes, UncleHeadersBytes, WithdrawalsBytes, DepositsBytes]
+  | [
+      BlockHeaderBytes,
+      TransactionsBytes,
+      UncleHeadersBytes,
+      WithdrawalsBytes,
+      DepositsBytes,
```
Comment (scorbajio) [inline packages/block/src/types.ts:186]: Not sure if this is going to be before or after `ExecutionWitnessBytes` in Prague.

Comment (holgerd77) [reply packages/block/src/types.ts:186]: ExecutionWitness is for Verkle which is not scheduled for Prague, so Deposits can always be placed before ExecutionWitness.

File: packages/common/src/eips.ts [hunk 2]
```diff
@@ -505,6 +505,13 @@ export const EIPs: EIPsDict = {
       },
     },
   },
+  6110: {
+    comment: 'Supply validator deposits on chain',
+    url: 'https://eips.ethereum.org/EIPS/eip-6110',
+    status: Status.Draft,
+    minimumHardfork: Hardfork.Cancun,
```
Comment (holgerd77) [inline packages/common/src/eips.ts:512]: The `minimumHardfork` is not the inclusion HF, but the first HF from which this EIP can be run in isolation (so: all pre-requisits for the EIP are met). So we can (should) switch to `Cancun` here, just as some additional context.

**Comment by scorbajio** [comment]
Implementation of EIP-6110 is being continued at #3390.

**Review by Smangele87 — COMMENTED** [review]
packages/common/src/eips.ts

**Pull request closed without merging**